mysite/ct/tests/integrate.py

```diff
@@ -0,0 +1,492 @@
+        l3.save_root(concept)
+        unit3 = Unit(title='Another Courselet', addedBy=self.user)
+        unit3.save()
+        ul3 = UnitLesson.create_from_lesson(l3, unit3)
+        clList = concept.get_conceptlinks(self.unit)  # should get l1, l2
+        self.assertEqual(len(clList), 2)
+        self.assertEqual([cl for cl in clList if cl.lesson == l1][0]
+                         .relationship, ConceptLink.DEFINES)
+        self.assertEqual([cl for cl in clList if cl.lesson == l2][0]
+                         .relationship, ConceptLink.TESTS)
+        self.assertEqual(distinct_subset([ul1, ul2, ul3]), [ul1, ul2])
+
+
+class LessonMethodTests(TestCase):
+    def setUp(self):
+        self.user = User.objects.create_user(username='jacob', email='jacob@_',
+                                             password='top_secret')
+        self.ul = create_question_unit(self.user)
+        concept = Concept.new_concept('bad', 'idea', self.ul.unit, self.user)
+        self.ul.lesson.concept = concept
+        self.ul.lesson.save()
+        self.unit2 = Unit(title='My Courselet', addedBy=self.user)
+        self.unit2.save()
+
+    def test_creation_treeID(self):
+        'treeID properly initialized to default?'
+        lesson = Lesson(title='foo', text='bar', addedBy=self.user)
+        lesson.save_root()
+        l2 = Lesson.objects.get(pk=lesson.pk)
+        self.assertEqual(l2.treeID, l2.pk)
+
+    def test_checkout(self):
+        'check Lesson commit and checkout'
+        self.assertFalse(self.ul.lesson.is_committed())
+        self.ul.lesson.conceptlink_set.create(concept=self.ul.lesson.concept,
+                                              addedBy=self.user)
+        ul2 = self.ul.copy(self.unit2, self.user)  # copy to new unit
+        self.assertTrue(self.ul.lesson.is_committed())
+        self.assertEqual(self.ul.lesson, ul2.lesson)
+        self.assertEqual(self.ul.lesson.changeLog, 'snapshot for fork by jacob')
+        self.assertNotEqual(self.ul.pk, ul2.pk)
+        self.assertEqual(ul2.lesson.title, self.ul.lesson.title)
+        lesson = ul2.checkout(self.user)
+        self.assertEqual(lesson.title, self.ul.lesson.title)
+        lesson.text = 'Big bad wolf'
+        lesson.changeLog = 'why I changed everything'
+        ul2.checkin(lesson)
+        ul2b = UnitLesson.objects.get(pk=ul2.pk)
+        self.assertEqual(ul2b.lesson.title, self.ul.lesson.title)
+        self.assertEqual(ul2b.lesson.concept, self.ul.lesson.concept)
+        self.assertEqual(ul2b.lesson.text, 'Big bad wolf')
+        self.assertFalse(self.ul.lesson.pk == ul2b.lesson.pk)
+        self.assertEqual(Concept.objects.get(conceptlink__lesson__unitlesson=ul2b),
+                         self.ul.lesson.concept)
+        self.assertTrue(ul2b.lesson.is_committed())
+
+
+class FakeRequest(object):
+    'trivial holder for request data to pass to test calls'
+    def __init__(self, user, sessionDict=None, method='POST', dataDict=None,
```

Comment:
Instead checking for `sessionDict ` for `None`, set defualt value to `sessionDict={}`